src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -7077,18 +7086,21 @@ instruct loadConD(vRegD dst, immD con) %{
 %}
 
 // Load Half Float Constant
-// The "ldr" instruction loads a 32-bit word from the constant pool into a
-// 32-bit register but only the bottom half will be populated and the top
-// 16 bits are zero.
 instruct loadConH(vRegF dst, immH con) %{
   match(Set dst con);
-  format %{
-    "ldrs $dst, [$constantaddress]\t# load from constant table: half float=$con\n\t"
-  %}
+  format %{ "mov    rscratch1, $con\n\t"
+            "fmov   $dst, rscratch1"
+         %}
   ins_encode %{
-    __ ldrs(as_FloatRegister($dst$$reg), $constantaddress($con));
+    uint32_t imm = (uint32_t)$con$$constant;
+    if (imm == 0) {
+      __ movw(rscratch1, zr);
+    } else {
+      __ movw(rscratch1, imm);
+    }
```

Comment:
Is this a Neoverse-specific optimization? On Apple M1, `mov x0, #0` is handled by renaming (so never issues) but `mov x0, xzr` is not eliminated. Let's go for the simplest here, this is too fussy.

```suggestion
    uint32_t imm = (uint32_t)$con$$constant;

    __ movw(rscratch1, imm);
```

Reply:
Could you please take another look at the patch? Thanks!